### PR TITLE
refactor: shuffle encodings around in Read Buffer

### PR DESCRIPTION
**Pull request by e-dard**
Not much to say here. No behaviour changes, I'm just trying to better organise the encodings in the Read Buffer as I prepare to add more.

I'm going to be organising them by logical type (string, scalar, bool etc) then by the type of encoding ("nothing", dictionary, RLE) etc.

### CODE REVIEW

File: read_buffer/src/column/encoding/string.rs [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
 use std::collections::BTreeSet;
 
 use either::Either;
 
 // This makes the encoding types available under the dictionary module.
-pub use self::plain::Plain;
-pub use self::rle::RLE;
+pub use dictionary::Dictionary;
+pub use rle::RLE;
```
Comment (tustvold) [inline read_buffer/src/column/encoding/string.rs:10]: RLE is still dictionary encoded no, perhaps it should be RLEDictionary or something?

File: read_buffer/benches/string.rs [hunk 2]
```diff
@@ -17,16 +17,16 @@ enum Location {
 }
 
 enum EncType {
-    RleDictionary,
+    Rle,
```
Comment (tustvold) [inline read_buffer/benches/string.rs:20]: I'm not sure I understand this, it isn't just RLE, it is dictionary encoded RLE?

**Review by tustvold — APPROVED** [review]
I don't feel strongly about this, so happy for you to go ahead with it, but if anything I feel the standardisation should standardise in the opposite direction :laughing:

File: read_buffer/benches/string.rs (hunk 2 above)
Comment (e-dard) [reply read_buffer/benches/string.rs:20]: The way I'm thinking about it at the moment is that it's RLE on strings, of which the implementation uses a disctionary to encode the run-lengths. But other RLE encodings in the RB may not use dictionaries. 

I think I'm going to go ahead and when I've added more encodings see if I should reorganise further or back 😄

**Pull request merged**